src/nvm.rs

```diff
@@ -307,26 +328,36 @@ impl<T, const N: usize> Collection<T, N> where T: Copy {
         self.capacity() - self.len()
     }
 
-    /// Returns the index of an item in the internal storage, given the index
-    /// in the collection. If index is too big, None is returned.
+    /// Counts the number of allocated slots up until `len`.
+    fn count_allocated(&self, len: usize) -> usize {
+        self.flags.get_ref()
+            .iter()
+            .take(len)
+            .fold(0, |acc, &byte| acc + (byte == STORAGE_VALID) as u32) as usize
+    }
```

Comment:
I initially had written
```rust
self.flags.get_ref()
    .iter()
    .filter(|&byte| *byte == STORAGE_VALID)
    .count()
```
but after digging on the interwebs it turns out [this weird casting is more performant](https://docs.rs/bytecount/0.6.2/src/bytecount/naive.rs.html#11-13)